wremnants/datasets/datagroups.py

```diff
@@ -172,32 +184,69 @@ def loadHistsForDatagroups(
         nominalIfMissing=True, applySelection=True, forceNonzero=True, pseudodata=False,
         preOpMap={}, preOpArgs={}, scaleToNewLumi=-1
     ):
+        logger.debug("Calling loadHistsForDatagroups()")
         logger.debug(f"the basename and syst is: {baseName}, {syst}")
         logger.debug(f"The procsToRead and excludedProcs are: {procsToRead}, {excluded_procs}")
         if self.rtfile and self.combine:
             self.setHistsCombine(baseName, syst, channel, procsToRead, excluded_procs, label)
         else:
-            self.setHists(baseName, syst, procsToRead, label, nominalIfMissing, applySelection, forceNonzero, preOpMap, preOpArgs, scaleToNewLumi=scaleToNewLumi)
-
-    def getDatagroups(self, excluded_procs=[]):
+            self.setHists(baseName, syst, procsToRead, label, nominalIfMissing, applySelection,
+                          forceNonzero, preOpMap, preOpArgs,
+                          scaleToNewLumi=scaleToNewLumi, excludeProcs=excluded_procs)
+
+    def addGroups(self, dictToAdd, canReplaceKey=False):
+        # dictToAdd is of the form {key: dict}, where dict is another dictionary with the group's specimen
+        for k in dictToAdd.keys():
+            if canReplaceKey or k not in self.groups.keys():
+                if k in self.groups.keys():
+                    logger.warning(f"Replacing {k} in groups")
+                self.groups[k] = dictToAdd[k]
+                if k not in self.groupNamesPostFilter:
+                    self.groupNamesPostFilter.append(k)
+            
+    def deleteGroup(self, procs):
+        for p in procs:
+            if p in self.groups.keys():
+                del self.groups[p]
+            if p in self.groupNamesPostFilter:
+                self.groupNamesPostFilter.remove(p)
+            
+    def getDatagroups(self, excluded_procs=[], afterFilter=False):
```

Comment:
I don't understand why we want to keep this "afterFilter" argument around and why the default would be false

Reply:
I agree, I had set it to false to maintain the previous behaviour until this was fully reviewed, but it turns out we always want it True, so I would actually set the default accordingly (I would still keep it thought, since it might happen that for some applications you might want to be able to access the full list of processes before the filters)

Reply:
Ok, yes, I think it should be true by default.